orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/CompoundExecutionOperator.java

```diff
@@ -0,0 +1,113 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline;
+
+import com.netflix.spinnaker.kork.core.RetrySupport;
+import com.netflix.spinnaker.orca.pipeline.model.Execution.ExecutionType;
+import com.netflix.spinnaker.orca.pipeline.persistence.ExecutionRepository;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import java.time.Duration;
+import javax.annotation.Nullable;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class CompoundExecutionOperator {
+  private ExecutionRepository repository;
+  private ExecutionRunner runner;
+  private RetrySupport retrySupport;
+
+  public CompoundExecutionOperator(
+      ExecutionRepository repository, ExecutionRunner runner, RetrySupport retrySupport) {
+    this.repository = repository;
+    this.runner = runner;
+    this.retrySupport = retrySupport;
+  }
+
+  public void cancel(ExecutionType executionType, String executionId) {
+    cancel(
+        executionType,
+        executionId,
+        AuthenticatedRequest.getSpinnakerUser().orElse("anonymous"),
+        null);
+  }
+
+  public void cancel(ExecutionType executionType, String executionId, String user, String reason) {
+    doInternal(
+        () -> runner.cancel(repository.retrieve(executionType, executionId), user, reason),
+        () -> repository.cancel(executionType, executionId, user, reason),
+        "cancel",
+        executionType,
+        executionId);
+  }
+
+  public void delete(ExecutionType executionType, String executionId) {
+    repository.delete(executionType, executionId);
+  }
+
+  public void pause(
+      @NonNull ExecutionType executionType,
+      @NonNull String executionId,
+      @Nullable String pausedBy) {
+    doInternal(
+        () -> runner.reschedule(repository.retrieve(executionType, executionId)),
+        () -> repository.pause(executionType, executionId, pausedBy),
+        "pause",
+        executionType,
+        executionId);
+  }
+
+  public void resume(
+      @NonNull ExecutionType executionType,
+      @NonNull String executionId,
+      @Nullable String user,
+      @NonNull Boolean ignoreCurrentStatus) {
+    doInternal(
+        () -> runner.unpause(repository.retrieve(executionType, executionId)),
+        () -> repository.resume(executionType, executionId, user, ignoreCurrentStatus),
+        "resume",
+        executionType,
+        executionId);
+  }
+
+  private void doInternal(
+      Runnable runnerAction,
+      Runnable repositoryAction,
+      String action,
+      ExecutionType executionType,
+      String executionId) {
+    try {
+      retrySupport.retry(
```

Comment:
why do you need the retry here? I didn't see it on the controller side..?
If you get rid of it, you can pass in a single lambda/runnable instead of two

Reply:
I thought it made sense to add, especially in a somewhat central place like this. Do you think it's unnecessary?

Reply:
all the runner does is push a thing onto the queue... Also, from this methods perspective it's just a runnable so it seems weird to distinguish one runnable and retry it but not the other one. I think it would be cleaner more obvious what's going on if a) there is one runnable and (optionally) b) it's either all retried or not

Reply:
It is weird that the repo action is not retried, I think it is more consistent in 9da1e69f4579ad12166e9a8db59f67d78af941ab.

I kept the 2 runnables, because I don't want to retry updating the queue if the repo action fails. Seems reasonable?